### PR TITLE
bug: Fix for getScreenshot returning empty image when using an external render target

### PR DESCRIPTION
**Pull request by astrid-kg**
#### Type of change

Bug 

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/REV-643

## Description :pencil:

The core issue here is of course that we render to a different render target, which means that the canvas we use to create screenshots is still empty when the image is created. 

This worked before because when setRenderTarget is called on a WebGLRenderer with null, it will fallback to use the canvas. We also did two renders, both on the renderer and the revealManager:

```
this.renderer.render(this._sceneHandler.scene, screenshotCamera);
this.revealManager.render(screenshotCamera);
```

This is wrong, as the one done directly on the renderer does not execute the render pipeline set up in RevealManager. However, this made the screenshot (kind of) work as the first render would render to the correct canvas. 

Now that we only render through the RevealManager, it came to light that changing the render target directly on the renderer does not affect the `DefaultRenderPipelineProvider`, as this will override whatever target has been set with its own `_outputRenderTarget`. Most of this PR is therefore about exposing this target so that it can be changed by users.

## How has this been tested? :mag:

I used Reveal Examples, and created a custom render target and passed this with the view options. I have build versions with `autoSetSize` set to both true and false:

```
      const customRenderTarget = new THREE.WebGLRenderTarget(300, 300);
      customRenderTarget.depthTexture = new THREE.DepthTexture(300, 300);
      
      [...]
      
        renderTargetOptions: {
          target: customRenderTarget,
          autoSetSize: false
        }
```

I then took several screenshots with default and custom resolutions, with and with and without UI included, at different viewer sizes and using different rendering modes. I have tested with Chrome, Firefox and on Android. I also tested a few screenshots without a custom render target to make sure the original feature did not break.

## Test instructions :information_source:

See above.

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [x] I have asked peers to test this feature.

### CODE REVIEW

**Review by Savokr — APPROVED** [review]
Looks good, a few comments but works fine as is. Tested in safari and chrome MacBook M1, works fine in chrome but not working in safari (both standard render target and custom) with webgl metal backend (which is default settings). Measurement labels are rendered but model and reveal window itself are not, just white background.

**Review by Savokr — CHANGES_REQUESTED** [review]
Found one bug, when `includeUI` flag is set to false in examples UI, Reveal resolution is reduced and not set to initial value before taking screenshot. Video available here: https://cognitedata.atlassian.net/browse/REV-643?focusedCommentId=232934

File: viewer/packages/api/src/public/migration/Cognite3DViewer.ts [hunk 1]
```diff
@@ -1183,6 +1184,11 @@ export class Cognite3DViewer {
       // Disregard pixelRatio to get the screenshot in requested resolution.
```
[the remark this answers is not in the document]
Comment (larsmoa) [reply viewer/packages/api/src/public/migration/Cognite3DViewer.ts:1184]: Although that would be good, I think its might be hard to do @Savokr - a new canvas means a new WebGL context which means that everything must be re-initialized.

File: viewer/packages/api/src/public/migration/Cognite3DViewer.ts [hunk 2]
```diff
@@ -1183,6 +1184,11 @@ export class Cognite3DViewer {
...
       const pixelRatioOverride = 1;
       this.renderer.setDrawingBufferSize(width, height, pixelRatioOverride);
+      if (customRenderTarget) {
+        // To esure that the render is done to the screenshot canvas, we clear the custom
+        // render target. This will make the renderer fall back to use the default canvas.
+        this.revealManager.setOutputRenderTarget(null);
```
[the remark this answers is not in the document]
Comment (larsmoa) [reply viewer/packages/api/src/public/migration/Cognite3DViewer.ts:1190]: This is not something we need to support.

File: viewer/packages/api/src/public/migration/Cognite3DViewer.ts [hunk 3]
```diff
@@ -1183,6 +1184,11 @@ export class Cognite3DViewer {
       // Disregard pixelRatio to get the screenshot in requested resolution.
       const pixelRatioOverride = 1;
       this.renderer.setDrawingBufferSize(width, height, pixelRatioOverride);
+      if (customRenderTarget) {
+        // To esure that the render is done to the screenshot canvas, we clear the custom
```
[the remark this answers is not in the document]
Comment (astrid-kg) [reply viewer/packages/api/src/public/migration/Cognite3DViewer.ts]: Done!

**Review by Savokr — APPROVED** [review]
Looks good, tested in safari and chrome, works just fine!

**Pull request merged**